api/src/test/java/org/pmiops/workbench/api/UserControllerTest.java

```diff
@@ -0,0 +1,207 @@
+  }
+
+  @Test
+  public void testUserPageSize() {
+    int size = 1;
+    UserResponse robinsons_0 = userController.user("Robinson", "0", size, null).getBody();
+    UserResponse robinsons_1 = userController.user("Robinson", "1", size, null).getBody();
+    UserResponse robinsons_2 = userController.user("Robinson", "2", size, null).getBody();
+    UserResponse robinsons_3 = userController.user("Robinson", "3", size, null).getBody();
+    UserResponse robinsons_4 = userController.user("Robinson", "4", size, null).getBody();
+
+    assertThat(robinsons_0.getUsers()).hasSize(size);
+    assertThat(robinsons_0.getNextPageToken()).isEqualTo("1");
+    assertThat(robinsons_1.getUsers()).hasSize(size);
+    assertThat(robinsons_1.getNextPageToken()).isEqualTo("2");
+    assertThat(robinsons_2.getUsers()).hasSize(size);
+    assertThat(robinsons_2.getNextPageToken()).isEqualTo("3");
+    assertThat(robinsons_3.getUsers()).hasSize(size);
+    assertThat(robinsons_3.getNextPageToken()).isEqualTo("4");
+    assertThat(robinsons_4.getUsers()).hasSize(size);
+    assertThat(robinsons_4.getNextPageToken()).isEqualTo("");
+  }
+
+  @Test
+  public void testUserPagedResponses() {
+    UserResponse robinsons_0_1 = userController.user("Robinson", "0", 2, null).getBody();
+    UserResponse robinsons_2_3 = userController.user("Robinson", "1", 2, null).getBody();
+    UserResponse robinsons_4 = userController.user("Robinson", "3", 1, null).getBody();
+
+    // Assert the expected size for each page
+    assertThat(robinsons_0_1.getUsers()).hasSize(2);
+    assertThat(robinsons_2_3.getUsers()).hasSize(2);
+    assertThat(robinsons_4.getUsers()).hasSize(1);
+
+    // Assert uniqueness across pages
+    assertThat(robinsons_0_1.getUsers()).containsNoneOf(robinsons_2_3, robinsons_4);
+    assertThat(robinsons_2_3.getUsers()).containsNoneOf(robinsons_0_1, robinsons_4);
+    assertThat(robinsons_4.getUsers()).containsNoneOf(robinsons_0_1, robinsons_2_3);
+  }
+
+  @Test
+  public void testUserSort() {
+    UserResponse robinsonsAsc = userController.user("Robinson", null, null, "asc").getBody();
+    UserResponse robinsonsDesc = userController.user("Robinson", null, null, "desc").getBody();
+
+    // Assert we have the same elements in both responses
+    assertThat(robinsonsAsc.getUsers()).containsAllIn(robinsonsDesc.getUsers());
+
+    // Now reverse one and assert both in the same order
+    List<org.pmiops.workbench.model.User> descendingReversed = Lists.reverse(robinsonsDesc.getUsers());
+    assertThat(robinsonsAsc.getUsers()).containsAllIn(descendingReversed).inOrder();
+
+    // Test that JPA sorting is really what we expected it to be by re-sorting one into a new list
+    List<org.pmiops.workbench.model.User> newAscending = Lists.newArrayList(robinsonsAsc.getUsers());
+    newAscending.sort(Comparator.comparing(org.pmiops.workbench.model.User::getEmail));
+    assertThat(robinsonsAsc.getUsers()).containsAllIn(newAscending).inOrder();
+  }
+
```

Comment:
Consider adding some test cases to cover invalid page tokens, including well-formed, but negative offset values.

Reply:
Done.